src/Routes.js

```diff
@@ -52,30 +53,17 @@ function checkPaths (routes, app) {
  *      path - https://prod.foo.redhat.com:1337/insights/advisor/rules
  *      component - component to be rendered when a route has been chosen.
  */
-export const Routes = ({ childProps: { history }}) => {
-    const pathName = window.location.pathname.split('/');
-    pathName.shift();
-
-    if (pathName[0] === 'beta') {
-        pathName.shift();
-    }
-
-    if (!checkPaths(paths, pathName[0])) {
-        history.push(paths.actions);
-    }
+export const Routes = (props: Props) => {
+    const path = props.childProps.location.pathname;
+    const paths = Object.values(pathsMap);
 
     return (
         <Switch>
-            <InsightsRoute path={ paths.actions } component={ Actions } rootClass='actions'/>
-            <InsightsRoute path={ paths.rules } component={ Rules } rootClass='rules'/>
+            <InsightsRoute path={ pathsMap.samplepage } component={ Actions } rootClass='actions'/>
+            <InsightsRoute path={ pathsMap.rules } component={ Rules } rootClass='rules'/>
+
+            { /* Finally, catch all unmatched routes */ }
+            <Route render={ () => some(paths, p => p === path) ? null : (<Redirect to={ paths.actions }/>) }/>
```

Comment:
This should work as replacement for `checkPaths` it needs to be tested to see if bookmarking still works.